Sources/Base/OAuth2Base.swift

```diff
@@ -451,6 +452,36 @@ open class OAuth2Base: OAuth2Securable {
 	*/
 	open func assureRefreshTokenParamsAreValid(_ params: OAuth2JSON) throws {
 	}
+	
+	
+	// MARK: - PKCE
+	
+	func generateCodeVerifier() -> String {
```

Comment:
Can you add unit test for it?

Reply:
Done - added top-level PKCE unit tests to OAuth2CodeGrantTests